data-node/src/main/java/org/graylog/datanode/Configuration.java

```diff
@@ -254,7 +254,10 @@ public class Configuration {
     @Parameter(value = "metrics_policy")
     private String metricsPolicy = "gl-datanode-metrics-ism";
 
-    @Documentation(value = "Cache size for searchable snaphots")
+    /**
+     * @see <a href="https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot/#configuring-a-node-to-use-searchable-snapshots}">Searchable snapshots</a>
+     */
+    @Documentation(value = "Cache size for searchable snaphots. ")
     @Parameter(value = "node_search_cache_size")
```

Comment:
would it make sense to use a `NamedBindingOverride` getter here to configure it to a different value if no s3 repo is configured?

Reply:
There are two types of searchable snaphsot repos - s3 and shared filesystem. Looking at it now, we could automatically  minimize the cache size based on either s3 configuration presence OR `path_repo` configured. If users want to use snapshots, they need to give us one or the other. So we could take it as an indicator that the cache is needed :thinking:

Reply:
yes, that sounds good. i will change the pr back to requested changes

Reply:
refactored, now we check both s3 and path_repo config and based on that decide if we add the search role and configure the cache.